source/Body.cpp

```diff
@@ -297,9 +297,27 @@ void Body::SetSwizzle(int swizzle)
 
 
 
-double Body::Alpha() const
+double Body::Alpha(const Point &drawCenter) const
 {
-	return alpha;
+	return alpha * DistanceAlpha(drawCenter);
+}
+
+
+
+double Body::DistanceAlpha(const Point &drawCenter) const
+{
+	if(!distanceInvisible)
+		return 1.;
+	double distance = (drawCenter - position).Length();
+	double distanceAlpha = max(min((distance - distanceInvisible) / (distanceVisible - distanceInvisible), 1.), 0.);
```

Comment:
You can probably use [clamp](https://en.cppreference.com/w/cpp/algorithm/clamp) here.

Reply:
```suggestion
	double distanceAlpha = clamp<double>((distance - distanceInvisible) / (distanceVisible - distanceInvisible), 0., 1.);
```